target/linux/ar71xx/patches-4.4/999-qca953x-clock-25mhz.patch

```diff
@@ -0,0 +1,62 @@
+From 941c5d4504552474188d36c02d466542ecfed6ac Mon Sep 17 00:00:00 2001
+From: Robert Marko <robimarko@gmail.com>
+Date: Sun, 5 Nov 2017 18:07:35 +0100
+Subject: [PATCH] CPE210 v2 ref clock
+---
+ arch/mips/ath79/clock.c    | 6 +++++-
+ arch/mips/ath79/dev-wmac.c | 7 +++----
+ 2 files changed, 8 insertions(+), 5 deletions(-)
+
+--- a/arch/mips/ath79/clock.c
++++ b/arch/mips/ath79/clock.c
+@@ -24,6 +24,7 @@
+ #include <asm/mach-ath79/ath79.h>
+ #include <asm/mach-ath79/ar71xx_regs.h>
+ #include "common.h"
++#include "machtypes.h"
+ 
+ #define AR71XX_BASE_FREQ	40000000
+ #define AR724X_BASE_FREQ	40000000
+@@ -365,9 +366,12 @@ static void __init qca953x_clocks_init(v
+ 	u32 bootstrap;
+ 
+ 	bootstrap = ath79_reset_rr(QCA953X_RESET_REG_BOOTSTRAP);
+-	if (bootstrap &	QCA953X_BOOTSTRAP_REF_CLK_40)
+-		ref_rate = 40 * 1000 * 1000;
+-	else
++	if (bootstrap &	QCA953X_BOOTSTRAP_REF_CLK_40) {
++		if (mips_machtype == ATH79_MACH_CPE210_V2)
++			ref_rate = 25 * 1000 * 1000;
++		else
++			ref_rate = 40 * 1000 * 1000;
++	} else
+ 		ref_rate = 25 * 1000 * 1000;
+ 
+ 	pll = ath79_pll_rr(QCA953X_PLL_CPU_CONFIG_REG);
```

Comment:
might be cleaner code to write 

if (mips_machtype == ATH79_MACH_CPE210_V2)
...
else if (bootstrap &	QCA953X_BOOTSTRAP_REF_CLK_40) 
...
else
...